dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -750,56 +854,72 @@ private void setHTMLPagesDependencies(Set<String> idsToWork) {
+			for (final String id : templatesSet) {
+				final Template workingTemplate = APILocator.getTemplateAPI().findWorkingTemplate(id, user, false);
+				final Template liveTemplate = APILocator.getTemplateAPI().findLiveTemplate(id, user, false);
 
 				// Host dependency
-				Host h = APILocator.getHostAPI().find(APILocator.getTemplateAPI().getTemplateHost(wkT).getIdentifier(), user, false);
-				hosts.addOrClean( APILocator.getTemplateAPI().getTemplateHost( wkT ).getIdentifier(), h.getModDate());
+				if (publisherFilter.acceptExcludeDependencyClasses(PusheableAsset.SITE.getType())) {
+					final Host host = APILocator.getHostAPI()
+							.find(APILocator.getTemplateAPI().getTemplateHost(workingTemplate).getIdentifier(),
+									user, false);
+					hosts.addOrClean(
+							APILocator.getTemplateAPI().getTemplateHost(workingTemplate).getIdentifier(),
+							host.getModDate());
+				}
 
 				containerList.clear();
-				containerList.addAll(APILocator.getTemplateAPI().getContainersInTemplate(wkT, user, false));
+				// Container dependencies
+				if (publisherFilter.acceptExcludeDependencyClasses(PusheableAsset.CONTAINER.getType())) {
+					containerList.addAll(APILocator.getTemplateAPI()
+							.getContainersInTemplate(workingTemplate, user, false));
 
-				if(lvT!=null && InodeUtils.isSet(lvT.getInode())) {
-					containerList.addAll(APILocator.getTemplateAPI().getContainersInTemplate(lvT, user, false));
-				}
+					if (liveTemplate != null && InodeUtils.isSet(liveTemplate.getInode())) {
+						containerList.addAll(APILocator.getTemplateAPI()
+								.getContainersInTemplate(liveTemplate, user, false));
+					}
 
-				for (Container container : containerList) {
+					for (final Container container : containerList) {
 
-					if(container instanceof FileAssetContainer){
-						fileAssetContainersSet.add(container.getIdentifier());
-						continue;
-					}
+						if (container instanceof FileAssetContainer) {
+							fileAssetContainersSet.add(container.getIdentifier());
+							continue;
+						}
 
-					// Container dependencies
-					containers.addOrClean( container.getIdentifier(), container.getModDate());
-					containersSet.add(container.getIdentifier());
+						containers.addOrClean(container.getIdentifier(), container.getModDate());
+						containersSet.add(container.getIdentifier());
+					}
 				}
 
 				//Adding theme
-				if(UtilMethods.isSet(wkT.getTheme())){
-					try{
-						Folder themeFolder = folderAPI.find(wkT.getTheme(), user, false);
-						if(themeFolder != null &&  InodeUtils.isSet(themeFolder.getInode())){
-							Folder parent = APILocator.getFolderAPI().findParentFolder(themeFolder, user, false);
-							if(UtilMethods.isSet(parent)) {
-								folders.addOrClean( parent.getInode(), parent.getModDate());
-								foldersSet.add(parent.getInode());
+				if (publisherFilter.acceptExcludeDependencyClasses(PusheableAsset.FOLDER.getType())) {
+					if (UtilMethods.isSet(workingTemplate.getTheme())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)